pandas/_libs/tslibs/conversion.pyx

```diff
@@ -636,34 +636,41 @@ cdef inline int64_t[:] _tz_convert_dst(int64_t[:] values, tzinfo tz,
     """
     cdef:
         Py_ssize_t n = len(values)
-        Py_ssize_t i, pos
+        Py_ssize_t i
+        int64_t[:] pos
         int64_t[:] result = np.empty(n, dtype=np.int64)
         ndarray[int64_t] trans
         int64_t[:] deltas
         int64_t v
+        bint tz_is_local
 
-    if not is_tzlocal(tz):
+    tz_is_local = is_tzlocal(tz)
+
+    if not tz_is_local:
         # get_dst_info cannot extract offsets from tzlocal because its
         # dependent on a datetime
         trans, deltas, _ = get_dst_info(tz)
         if not to_utc:
             # We add `offset` below instead of subtracting it
             deltas = -1 * np.array(deltas, dtype='i8')
 
+    if not tz_is_local:
```

Comment:
Do we need to `if not tz_is_local` logic blocks here (lines 647 and 657)?

Reply:
Nope, will merge them in next push

Reply:
Done